sherpa/bin/conv_emformer_transducer_stateless2/beam_search.py

```diff
@@ -0,0 +1,254 @@
+        )
+
+        (
+            encoder_out,
+            encoder_out_lens,
+            next_states,
+        ) = model.encoder_streaming_forward(
+            features=features,
+            features_length=features_length,
+            num_processed_frames=num_processed_frames,
+            states=states,
+        )
+
+        processed_lens = (num_processed_frames >> 2) + encoder_out_lens
+        next_hyp_list = fast_beam_search_one_best(
+            model=model,
+            encoder_out=encoder_out,
+            processed_lens=processed_lens,
+            rnnt_decoding_config=rnnt_decoding_config,
+            rnnt_decoding_streams_list=rnnt_decoding_streams_list,
+        )
+
+        next_state_list = unstack_states(next_states)
+        for i, s in enumerate(stream_list):
+            s.states = next_state_list[i]
+            s.hyp = next_hyp_list[i]
+
+    def get_texts(self, stream):
+        """
+        Return text after decoding
+        Args:
+          stream:
+            Stream to be processed.
+        """
+        return self.sp.decode(stream.hyp[self.context_size :])
+
+
+class GreedySearch:
+    def __init__(self, model, device):
+        """
+        Args:
+          model:
+            RNN-T model decoder model
+          device:
+            Device on which the computation will occur
+        """
+
+        self.blank_id = model.blank_id
+        self.context_size = model.context_size
+        self.device = device
+
+        decoder_input = torch.tensor(
+            [[self.blank_id] * self.context_size],
+            device=self.device,
+            dtype=torch.int64,
+        )
+        initial_decoder_out = model.decoder_forward(decoder_input)
+        self.initial_decoder_out = model.forward_decoder_proj(
+            initial_decoder_out.squeeze(1)
+        )
+
+    def init_stream(self, stream: "Stream"):
+        """
+        Attributes to add to each stream
+        """
+        stream.decoder_out = self.initial_decoder_out
+        stream.hyp = [self.blank_id] * self.context_size
+
+    @torch.no_grad()
+    def process(
+        self,
+        server: "StreamingServer",
+        stream_list: List[Stream],
+    ) -> None:
+        """Run the model on the given stream list and do search with given decoding
+           method.
+        Args:
+          server:
+            An instance of `StreamingServer`.
+          stream_list:
+            A list of streams to be processed. It is changed in-place.
+            That is, the attribute `states`, `decoder_out`, and `hyp` are
+            updated in-place.
+        """
+        model = server.model
+        device = model.device
+        # Note: chunk_length is in frames before subsampling
+        chunk_length = server.chunk_length
+        batch_size = len(stream_list)
+        chunk_length_pad = server.chunk_length_pad
+        state_list, feature_list = [], []
+        decoder_out_list, hyp_list = [], []
+        processed_frames_list = []
```

Comment:
```suggestion
```

For `greedy_search` you don't need `processed_frames_list`.

It is used only for `fast_beam_search` based decoding methods.

Reply:
In ```icefall``` it seems like it is needed for prediction, isn't it ?

Reply:
Yes, I think it is needed by streaming encoder forward.